collect_app/src/main/java/org/odk/collect/android/widgets/BaseImageWidget.java

```diff
@@ -154,27 +157,25 @@ protected void addCurrentImageToLayout() {
             int screenHeight = metrics.heightPixels;
 
             File f = new File(getInstanceFolder() + File.separator + binaryName);
+            if (!f.exists()) {
+                f = addDefaultImageIfExists(f);
```

Comment:
I think that here you can just try to use the reference manager to resolve the ref and go with that if it works. You don't want to copy the image and you don't want to change what value is in the instance (`binaryName`). The instance value should only change if the image is replaced (including if an image previously from `jr://images/` is annotated).

Reply:
If I don't copy the image it won't be saved in instance dir and it won't be uploaded to a server.

Reply:
That is the intent! You can read more about that in the forum thread. You'll see I wasn't on board initially but I was convinced.

Reply:
Ok I fixed it, now it should be ok.